instrumentation_test.go

```diff
@@ -95,6 +96,16 @@ func TestWithEnv(t *testing.T) {
 		_, err = newInstConfig(ctx, opts)
 		require.ErrorContains(t, err, `parse log level "invalid"`)
 	})
+
+	// Test that autoexport.NewSpanExporter works when OTEL_TRACES_EXPORTER is
+	// not set
+	t.Run("With OTEL_TRACES_EXPORTER not set", func(t *testing.T) {
+		os.Unsetenv("OTEL_TRACES_EXPORTER")
+		c, err := newInstConfig(context.Background(), []InstrumentationOption{WithEnv()})
+
+		require.NoError(t, err)
+		assert.NotNil(t, c.traceExp)
```

Comment:
This should validate the instance type to be and exporter from `otlptracegrpc`. Given the structure of those exporters, calling [`MarshalLog`](https://pkg.go.dev/go.opentelemetry.io/otel/exporters/otlp/otlptrace#Exporter.MarshalLog) is likely your best option.

```suggestion
		require.NotNil(t, c.traceExp)
		require.IsType(t, &otlptrace.Exporter{}, c.traceExp)
		exp := c.traceExp.(*otlptrace.Exporter)

		var buf bytes.Buffer
		logger := stdr.New(log.New(&buf, "", log.LstdFlags))
		logger.Info("", "exporter", exp)
		got, err := io.ReadAll(&buf)
		require.NoError(t, err)
		// "otlphttphttp" should be "otlptracegrpc". This is a bug upstream.
		assert.Contains(t, string(got), "otlphttphttp")
```

Reply:
Will be impacted by https://github.com/open-telemetry/opentelemetry-go/pull/6143. We can follow up with that update later though.